### PR TITLE
Advisor 2022 10 01

### PR DESCRIPTION
**Pull request by uparameswara**
## ARM API Information (Control Plane)

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow.
</i>
Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [ ] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by uparameswara** [comment]
@weidongxu-microsoft Hi - have a question regarding the failures:
 Swagger SDK breaking change tracking
      The error message indicates the following:
                   
image
      But we do have skiptoken and top provided in our previous approved versions. Am confused on why this error is happening. Could you please let me know the mitigation steps from our side?

**Comment by weidongxu-microsoft** [comment]
> @weidongxu-microsoft Hi - have a question regarding the failures: Swagger SDK breaking change tracking The error message indicates the following:
> 
> image
> 
> But we do have skiptoken and top provided in our previous approved versions. Am confused on why this error is happening. Could you please let me know the mitigation steps from our side?

Add @qiliao123 for approval on JS breaking change.

The failure on Go will block the PR. Add @tadelesh if you need help.

**Comment by tadelesh** [comment]
@uparameswara Could you explain the difference between `advisorScoreEntity` and `scoreEntity`? I need to rename one of them to prevent Go SDK naming collision.

**Comment by uparameswara** [comment]
> @uparameswara Could you explain the difference between `advisorScoreEntity` and `scoreEntity`? I need to rename one of them to prevent Go SDK naming collision.

@tadelesh  We introduced AdvisorScoreEntity and ScoreEntity in this stable release. 

AdvisorScoreEntity gives the information about  a specific score with respect to an Advisor Category - the AdvisorScoreEntity uses the ScoreEntity to display the last refreshed score.

**Comment by uparameswara** [comment]
> > @weidongxu-microsoft Hi - have a question regarding the failures: Swagger SDK breaking change tracking The error message indicates the following:
> > image
> > But we do have skiptoken and top provided in our previous approved versions. Am confused on why this error is happening. Could you please let me know the mitigation steps from our side?
> 
> Add @qiliao123 for approval on JS breaking change.
> 
> The failure on Go will block the PR. Add @tadelesh if you need help.
> Thanks @weidongxu-microsoft  we already have the arm manifest deployed - @qiliao123  can you please assist me with the JS breaking change ?

**Comment by weidongxu-microsoft** [comment]
> we already have the arm manifest deployed

This is something unusually. I thought you'd need to be first get approved by ARM before they'd allow you to deploy manifest...

But that's ARM problem. I don't have concern.

**Comment by uparameswara** [comment]
@weidongxu-microsoft : Hi any other suggestions/specs before we can merge this PR?

**Comment by sjanamma** [comment]
@uparameswara, For review efficiency consideration, when creating a new api version, it is required to place API specs of the base version in the first commit, and push new version updates into successive commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version). Or you could onboard [API spec pipeline](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/606/Enable-API-Spec-Pipeline)

File: specification/advisor/resource-manager/Microsoft.Advisor/stable/2022-10-01/advisor.json
Comment (sjanamma) [inline specification/advisor/resource-manager/Microsoft.Advisor/stable/2022-10-01/advisor.json:30]: could you add the pattern for restriction?

File: specification/advisor/resource-manager/Microsoft.Advisor/stable/2022-10-01/advisor.json
Comment (sjanamma) [inline specification/advisor/resource-manager/Microsoft.Advisor/stable/2022-10-01/advisor.json:66]: We prefer to pluralize the segment for the list. Can you make it entities instead of metadata. This might need a change on the GET call above as well

File: specification/advisor/resource-manager/Microsoft.Advisor/stable/2022-10-01/advisor.json
Comment (sjanamma) [inline specification/advisor/resource-manager/Microsoft.Advisor/stable/2022-10-01/advisor.json:319]: Since you are returning 202, I presume this is a long running operation. You must have x-ms-long-running-operation and x-ms-long-running-operation-options extensions.

[https://github.com/Azure/azure-openapi-validator/blob/main/docs/long-running-operations-options-validator.md](https://github.com/Azure/azure-openapi-validator/blob/main/docs/long-running-operations-options-validator.md)

[https://github.com/Azure/azure-openapi-validator/blob/main/docs/lro-extension.md](https://github.com/Azure/azure-openapi-validator/blob/main/docs/lro-extension.md)

File: specification/advisor/resource-manager/Microsoft.Advisor/stable/2022-10-01/advisor.json
Comment (sjanamma) [inline specification/advisor/resource-manager/Microsoft.Advisor/stable/2022-10-01/advisor.json:362]: @zizw123, @rkmanda - Can you please advise on this? Is this endpoint required? Using generateRecommendations as a segment in the path doesn't make sense.

File: specification/advisor/resource-manager/Microsoft.Advisor/stable/2022-10-01/advisor.json
Comment (uparameswara) [reply specification/advisor/resource-manager/Microsoft.Advisor/stable/2022-10-01/advisor.json:30]: @sjanamma : Hello ! Am not sure about the ask here . Will reach out to you offline wrt to resolving this

**Comment by uparameswara** [comment]
> @uparameswara, For review efficiency consideration, when creating a new api version, it is required to place API specs of the base version in the first commit, and push new version updates into successive commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version). Or you could onboard [API spec pipeline](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/606/Enable-API-Spec-Pipeline)

@sjanamma thank you for the  above documentation - for this particular api version change; we wanted to introduce a new api path 
image

The rest of the api specs are similar to the last published  2022-09-01 version.  Would you need us to restart the PR with the above approach?

File: specification/advisor/resource-manager/Microsoft.Advisor/stable/2022-10-01/advisor.json
Comment (uparameswara) [reply specification/advisor/resource-manager/Microsoft.Advisor/stable/2022-10-01/advisor.json:66]: @sjanamma : The current 2022-10-01 manifest for arm has been deployed with the above path. Advisor will make sure to pluralize the above spec in the future arm / swagger updates. Tagging @jportugal0 for visibility

File: specification/advisor/resource-manager/Microsoft.Advisor/stable/2022-10-01/advisor.json
Comment (weidongxu-microsoft) [inline specification/advisor/resource-manager/Microsoft.Advisor/stable/2022-10-01/advisor.json:846]: Likely be x-ms-pageable

File: specification/advisor/resource-manager/Microsoft.Advisor/stable/2022-10-01/advisor.json
Comment (weidongxu-microsoft) [inline specification/advisor/resource-manager/Microsoft.Advisor/stable/2022-10-01/advisor.json:1714]: Is it a ProxyResource?

File: specification/advisor/resource-manager/Microsoft.Advisor/stable/2022-10-01/advisor.json
Comment (weidongxu-microsoft) [inline specification/advisor/resource-manager/Microsoft.Advisor/stable/2022-10-01/advisor.json:1646]: Give a detailed type, int or float.

File: specification/advisor/resource-manager/Microsoft.Advisor/stable/2022-10-01/advisor.json
Comment (weidongxu-microsoft) [inline specification/advisor/resource-manager/Microsoft.Advisor/stable/2022-10-01/advisor.json:1697]: Empty?

**Comment by weidongxu-microsoft** [comment]
Note that the change on x-ms-longrunning-operation would get you to the breaking changes review.

**Comment by uparameswara** [comment]
> Note that the change on x-ms-longrunning-operation would requrie the breaking changes review.

@weidongxu-microsoft  Thanks ! i will remove the x-ms-longrunning-operation for now ; so as to remain consistent with the previous stable version for the api path; as i have not changed it in this vesrion.

File: specification/advisor/resource-manager/Microsoft.Advisor/stable/2022-10-01/advisor.json
Comment (uparameswara) [reply specification/advisor/resource-manager/Microsoft.Advisor/stable/2022-10-01/advisor.json:842]: @weidongxu-microsoft : The advisor score returns is a list of 5 score categories : Cost, Performance,Security,Operational excellance, Reliability and is static- hence not tagging it with x-ms-pageable.

File: specification/advisor/resource-manager/Microsoft.Advisor/stable/2022-10-01/advisor.json
Comment (weidongxu-microsoft) [reply specification/advisor/resource-manager/Microsoft.Advisor/stable/2022-10-01/advisor.json:1642]: Ignore me, this seems indeed float.

File: specification/advisor/resource-manager/Microsoft.Advisor/stable/2022-10-01/advisor.json
Comment (uparameswara) [reply specification/advisor/resource-manager/Microsoft.Advisor/stable/2022-10-01/advisor.json:1710]: @weidongxu-microsoft :  as represented here: 
image

https://learn.microsoft.com/en-us/rest/api/advisor/advisorscore(preview)/advisor-score/list?tabs=HTTP

File: specification/advisor/resource-manager/Microsoft.Advisor/stable/2022-10-01/advisor.json
Comment (weidongxu-microsoft) [reply specification/advisor/resource-manager/Microsoft.Advisor/stable/2022-10-01/advisor.json:842]: Then why does response schema has `nextLink`?

File: specification/advisor/resource-manager/Microsoft.Advisor/stable/2022-10-01/advisor.json
Comment (weidongxu-microsoft) [reply specification/advisor/resource-manager/Microsoft.Advisor/stable/2022-10-01/advisor.json:1710]: I am asking whether the schema belongs to ARM RroxyResource.
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v4/types.json#L91

Let ARM decide.

File: specification/advisor/resource-manager/Microsoft.Advisor/stable/2022-10-01/advisor.json
Comment (uparameswara) [reply specification/advisor/resource-manager/Microsoft.Advisor/stable/2022-10-01/advisor.json:1689]: @weidongxu-microsoft : was following the convention specified here : 
image

File: specification/advisor/resource-manager/Microsoft.Advisor/stable/2022-10-01/advisor.json
Comment (uparameswara) [reply specification/advisor/resource-manager/Microsoft.Advisor/stable/2022-10-01/advisor.json:842]: Good point. Removed from the  schema

File: specification/advisor/resource-manager/Microsoft.Advisor/stable/2022-10-01/advisor.json
Comment (zizw123) [reply specification/advisor/resource-manager/Microsoft.Advisor/stable/2022-10-01/advisor.json:362]: This also seems confusing to me, please elaborate more on the use case here, and is this already in the previous API version? I see in this PR bump up API from 2022-09-01 to 2022-10-01

File: specification/advisor/resource-manager/Microsoft.Advisor/stable/2022-10-01/advisor.json
Comment (zizw123) [reply specification/advisor/resource-manager/Microsoft.Advisor/stable/2022-10-01/advisor.json:1706]: @uparameswara You should consider use common type whenever is possible

File: specification/advisor/resource-manager/Microsoft.Advisor/stable/2022-10-01/advisor.json
Comment (uparameswara) [reply specification/advisor/resource-manager/Microsoft.Advisor/stable/2022-10-01/advisor.json:362]: @zizw123 
The generate status api call above - gets the status for a recommendation generation process - be it a 202 or 204.
 This is from the previous stable version we had before (2022-09-01) . This was bumped up to the newer version 2022-10-01 ( no changes here with the 2022-09-01 version) ; 

The bump to 2022-10-01 was made to introduce advisor score api paths as listed below in the comments.

File: specification/advisor/resource-manager/Microsoft.Advisor/stable/2022-10-01/advisor.json
Comment (uparameswara) [reply specification/advisor/resource-manager/Microsoft.Advisor/stable/2022-10-01/advisor.json]: @zizw123 @weidongxu-microsoft  :  Hi, with the latest commit I pushed to generalize as proxyResource - it seems the  go and python builds broke for me 

image

Can you please review my latest commit and help to figure how i can possibly rectify the issue?

image

Comment (weidongxu-microsoft) [reply specification/advisor/resource-manager/Microsoft.Advisor/stable/2022-10-01/advisor.json]: You JSON defined another "Resource" there.
https://github.com/Azure/azure-rest-api-specs/blob/d42498e7692c3cd2270e67e3f7bec6828c341fa1/specification/advisor/resource-manager/Microsoft.Advisor/stable/2022-10-01/advisor.json#L1337-L1357

That's a tough decision. You may use the local one this way
https://github.com/Azure/azure-rest-api-specs/blob/d42498e7692c3cd2270e67e3f7bec6828c341fa1/specification/advisor/resource-manager/Microsoft.Advisor/stable/2022-10-01/advisor.json#L1031-L1035

Again I leave this to ARM review.

Comment (uparameswara) [reply specification/advisor/resource-manager/Microsoft.Advisor/stable/2022-10-01/advisor.json]: Removed the local instance of Resouce and replaced it with the common-types Resource. Thanks @weidongxu-microsoft  and @zizw123  for your prompt assistance :)

File: specification/advisor/resource-manager/Microsoft.Advisor/stable/2022-10-01/advisor.json
Comment (uparameswara) [reply specification/advisor/resource-manager/Microsoft.Advisor/stable/2022-10-01/advisor.json:30]: Thanks for the review - Advisor team has this comment taken as PBI to include the change suggested here : https://msazure.visualstudio.com/One/_backlogs/backlog/AzureAdvisor/Features/?workitem=17048171 and will introduce the change needed by the comment in the next api-version increment of advisor.

File: specification/advisor/resource-manager/Microsoft.Advisor/stable/2022-10-01/advisor.json
Comment (uparameswara) [reply specification/advisor/resource-manager/Microsoft.Advisor/stable/2022-10-01/advisor.json:66]: Thanks for the review - Advisor team has this comment taken as PBI to include the change suggested here : https://msazure.visualstudio.com/One/_backlogs/backlog/AzureAdvisor/Features/?workitem=17048171 and will introduce the change needed by the comment in the next api-version increment of advisor.

File: specification/advisor/resource-manager/Microsoft.Advisor/stable/2022-10-01/advisor.json
Comment (uparameswara) [reply specification/advisor/resource-manager/Microsoft.Advisor/stable/2022-10-01/advisor.json:319]: Thanks for the review - Advisor team has this comment taken as PBI to include the change suggested here : https://msazure.visualstudio.com/One/_backlogs/backlog/AzureAdvisor/Features/?workitem=17048171 and will introduce the change needed by the comment in the next api-version increment of advisor.

**Review by weidongxu-microsoft — APPROVED** [review]
@qiaozha for potential js SDK breaks
@msyyc for potential  python SDK breaks

**Comment by weidongxu-microsoft** [comment]
We need "API Readiness Check" to pass, to ensure your service is onboard ARM.

**Comment by weidongxu-microsoft** [comment]
@qiaozha for potential js SDK breaks

**Pull request merged**
